src/bluetooth.rs

```diff
@@ -537,33 +272,6 @@ impl BluetoothDiscoverySession {
 
 impl BluetoothDevice {
 
-    fn create_device(adapter: BluetoothAdapter, device: String) -> BluetoothDevice {
-        match adapter {
-            #[cfg(all(target_os = "linux", feature = "bluetooth"))]
-            BluetoothAdapter::Bluez(_bluez_adapter) => {
-                BluetoothDevice::Bluez(Arc::new(BluetoothDeviceBluez::new(device)))
-            },
-            #[cfg(all(target_os = "android", feature = "bluetooth"))]
-            BluetoothAdapter::Android(android_adapter) => {
-                BluetoothDevice::Android(Arc::new(BluetoothDeviceAndroid::new(android_adapter, device)))
-            },
-            #[cfg(all(target_os = "macos", feature = "bluetooth"))]
-            BluetoothAdapter::Mac(mac_adapter) => {
-                BluetoothDevice::Mac(Arc::new(BluetoothDeviceMac::new(mac_adapter, device)))
-            },
-            #[cfg(not(any(all(target_os = "linux", feature = "bluetooth"),
-                          all(target_os = "android", feature = "bluetooth"),
-                          all(target_os = "macos", feature = "bluetooth"))))]
-            BluetoothAdapter::Empty(_adapter) => {
-                BluetoothDevice::Empty(Arc::new(BluetoothDeviceEmpty::new(device)))
-            },
-            #[cfg(feature = "bluetooth-test")]
-            BluetoothAdapter::Mock(fake_adapter) => {
-                BluetoothDevice::Mock(FakeBluetoothDevice::new_empty(fake_adapter, device))
-            },
-        }
-    }
-
     #[cfg(feature = "bluetooth-test")]
     pub fn create_mock_device(adapter: BluetoothAdapter, device: String) -> Result<BluetoothDevice, Box<Error>> {
```

Comment:
This method will complain if you run `cargo build --features bluetooth-test`.

Reply:
I was not aware of where create_mock_device method is being called. I was thinking if I should remove it?

Reply:
https://github.com/servo/servo/blob/f1dd31f70440fa9c7a40525bd1e03eede568f74d/components/bluetooth/test.rs#L153

Reply:
Isn't this another API that uses 'adapter: BluetoothAdapater' as a parameter? So it expects a trait but, tries to match with enum variants. We faced a similar issue for BluetoothDevice and BluetoothDiscoverySession, and as you suggested, we inlined the functions in adapter.rs itself. So, should I write this function in adapter.rs itself and remove it from bluetooth.rs?

Reply:
Yes.